tests/NovelRT.Tests/Ecs/ConfiguratorTest.cpp

```diff
@@ -40,6 +40,31 @@ TEST(ConfiguratorTest, ConfiguratorCanProduceSystemSchedulerWithCustomComponentA
     EXPECT_EQ(scheduler.GetComponentCache().GetComponentBuffer<int32_t>().GetComponent(1), 20);
 }
 
+TEST(ConfiguratorTest, ConfiguratorCanProduceSystemSchedulerWithCustomComponentAndIEcsSystem)
+{
+    class TestSystem : public IEcsSystem
+    {
+    public:
+        void Update(NovelRT::Timing::Timestamp, Catalogue catalogue) final
+        {
+            auto intComponents = catalogue.GetComponentView<int32_t>();
+            for (auto [entity, component] : intComponents)
+            {
+                intComponents.PushComponentUpdateInstruction(entity, 10);
+            }
+        }
+    };
+
+    std::shared_ptr<TestSystem> tester = std::make_shared<TestSystem>();
```

Comment:
This could just be `auto` given the fact you're calling make shared. No reason to have type information twice unless you're explicitly changing said type I don't think.

Reply:
Done.